### ISSUE TITLE
Change observers not called for some changes

### ISSUE THREAD

**Issue by andrewdavey** [optimizely/nuclear-js#140]
I've hit a very weird bug in my app. Certain orderings of changes to my store result in change observers not being called.

I'm concerned that maybe I'm running into #86 

I've managed to create a minimal reproduction.

```js
var Nuclear = require("nuclear-js"); // v1.1.1

var SET_YEAR_GROUP = "SET_YEAR_GROUP";
var LOADED = "LOADED";

function runTest(value) {

  var store = Nuclear.Store({
    getInitialState: function() {
      return Nuclear.toImmutable({
        yearGroup: 0,
        shouldLoaded: true
      });
    },

    initialize: function() {
      this.on(SET_YEAR_GROUP, setYearGroup);
      this.on(LOADED, loaded);
    }
  });

  function setYearGroup(store, payload) {
    return store
      .set("yearGroup", payload.yearGroup)
      .set("shouldLoad", true);
  }

  function loaded(store) {
    return store.set("shouldLoad", false);
  }

  var reactor = Nuclear.Reactor();

  reactor.registerStores({ uiStore: store });

  var output = [];
  // Record changes to yearGroup
  reactor.observe(["uiStore", "yearGroup"], function(y) { output.push(y) });

  reactor.dispatch(SET_YEAR_GROUP, {yearGroup: 6});
  reactor.dispatch(LOADED);
  reactor.dispatch(SET_YEAR_GROUP, {yearGroup: value});

  return output;
}

var output1 = runTest(1);
console.log(output1); // OK - logs [ 6, 1 ]

var output5 = runTest(5);
console.log(output5); // WRONG! Logs [ 6 ] instead of [ 6, 5 ] !

// Removing the reactor.dispatch("LOADED"); call seems to fix things, but why?!
```

Is this a bug in Nuclear?

**Comment by jordangarcia** [comment]
hm, this may be a bug due to the evaluator that has been mentioned here: https://github.com/optimizely/nuclear-js/issues/86

I will take a look and see if i can repro myself.

Thanks for reporting this.

**Comment by andrewdavey** [comment]
Thanks Jordan. Let me know if there is anything I can help with. This issue will be blocking my app's release, so I'm motivated to get it fixed ;)

**Comment by andrewdavey** [comment]
Are there any work-arounds I can try? I'm nearing a release deadline and really need to have this issue resolved.

**Comment by bhamodi** [comment]
@jordangarcia

**Comment by balloob** [comment]
I have run into this issue too. The work around that I use is to store true
as a string. You can update the getter to do !!value to be backwards
compatible.

**Comment by jordangarcia** [comment]
@andrewdavey

I'll be into this today and hopefully rolling out a fix

**Comment by jordangarcia** [comment]
This will be fixed in 1.2

**Comment by bhamodi** [comment]
@jordangarcia fixed this in v1.2.0.
